models/buffer_disk.go

```diff
@@ -0,0 +1,196 @@
+package models
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/tidwall/wal"
+)
+
+type DiskBuffer struct {
+	BufferStats
+	sync.Mutex
+
+	walFile     *wal.Log
+	walFilePath string
```

Comment:
You might want to shorten the names e.g.
```suggestion
	file     *wal.Log
	path string
```